### PR TITLE
Bundle latest gio2 and gobject-introspection

### PR DESCRIPTION
**Pull request by kenhys**
Enable by --with-bundled-ruby-glib2.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,6 +23,14 @@
 /binding/ruby/glib-*/ext/glib2/glib2.so
 /binding/ruby/glib-*/ext/glib2/mkmf.log
 /binding/ruby/glib-*/ext/glib2/ruby-glib2.pc
```
Comment (kou) [inline .gitignore:25]: Can we remove `glib-*/` entries?

**Review by kou — APPROVED** [review]
It's difficult to review...
We can merge this.

File: .gitignore (hunk 1 above)
Comment (kenhys) [reply .gitignore]: Removed.

**Comment by kou** [comment]
Could you also bundle pkg-config gem and native-package-installer gem like test-unit gem (`binding/ruby/test-unit/`).
We can use `git submodule` for them like `binding/ruby/test-unit/`.
We don't need to install them because they are needed only on building glib2 gem, gobject-introspection gem and gio2 gem.

**Comment by kenhys** [comment]
I'll bundle them after merging this PR.

**Pull request merged**